inference-engine/src/mkldnn_plugin/nodes/bucketize.cpp

```diff
@@ -29,17 +29,23 @@ class BucketizeImpl : public ExtLayerBase {
             // check one attribute
             with_right = layer->GetParamAsBool("with_right_bound");
 
-            // check precisions for input tensors
-            Precision input_tensor_precision = layer->insData[INPUT_TENSOR_PORT].lock()->getTensorDesc().getPrecision();
-            if (input_tensor_precision != Precision::FP32) {
-                THROW_IE_EXCEPTION << layer->name << " Incorrect input precision of the input. Only FP32 is supported!";
+            // check precisions for input and output tensors
+            input_precision = layer->insData[INPUT_TENSOR_PORT].lock()->getTensorDesc().getPrecision();
```

Comment:
It is not correct use of weak ptr.

`layer->insData[INPUT_TENSOR_PORT].lock()` can return nullptr.

Please first of all lock the pointer and only after that work with it.

Reply:
Done